### ISSUE TITLE
smbus not found

### ISSUE THREAD

**Issue by AdrienLF** [pimoroni/button-shim#9]
Hi,

I'm using raspbian stretch lite and after entering the one line installer my simple script won't run, saying smbus is not installed. Here is the error message: 

`Traceback (most recent call last):
  File "/home/pi/Scripts/Seismograph/Button-shim-test.py", line 1, in <module>
    import buttonshim
  File "/usr/local/lib/python3.6/site-packages/buttonshim/__init__.py", line 1, in <module>
    import smbus
ModuleNotFoundError: No module named 'smbus'`

I looked online how to install smbus and I have conflicting answers. I tried

`sudo pip3 install smbus`

and I get 
` Could not find a version that satisfies the requirement smbus (from versions: )
No matching distribution found for smbus`

Which I don't really understand. Can you help me find a way around this issue? 

Thanks!

**Comment by Gadgetoid** [comment]
smbus is an odd package that's provided by apt, you should be able to fix this with:

```
sudo apt install python-smbus
```

**Comment by AdrienLF** [comment]
Unfortunately it says 

`python-smbus is already the newest version (3.1.2-3).
`

Which is weird. Is it a case of it's installed for python2 and not 3?

**Comment by Gadgetoid** [comment]
Which version of Python 3 are you running? I suppose it's possible that the package is installing for (for example) 3.4, but version 3.6 is running. This is just a guess, though!

Might be worth looking what files the `python3-smbus` package installs, which you can do with `dpkg -L python3-smbus` and look to see if it's generic python3 or specific to one version.

**Comment by AdrienLF** [comment]
It may be the issue yes, as I have python 3.6 installed. Here's what I get when I do `dpkg -L python3-smbus`

/.
/usr
/usr/lib
/usr/lib/python3
/usr/lib/python3/dist-packages
/usr/lib/python3/dist-packages/smbus.cpython-35m-arm-linux-gnueabihf.so
/usr/share
/usr/share/doc
/usr/share/doc/python3-smbus
/usr/share/doc/python3-smbus/changelog.Debian.gz
/usr/share/doc/python3-smbus/changelog.gz
/usr/share/doc/python3-smbus/copyright

My python3 is in `/usr/local/bin/python3`. I expected smbus to be in the python3 folder, but maybe I am wrong? 
I also just received the envirophat and have the same issue with smbus.

**Comment by Gadgetoid** [comment]
This may be the problem: `/usr/lib/python3/dist-packages/smbus.cpython-35m-arm-linux-gnueabihf.so`

It's a C extension to Python which is explicitly stating compatibility with Python 3.5, I don't believe Python 3.6 will load it. Though I don't recall any major differences between the 3.5 and 3.6 C API.

You could try a quick hack:

```
sudo ln -s /usr/lib/python3/dist-packages/smbus.cpython-35m-arm-linux-gnueabihf.so /usr/lib/python3/dist-packages/smbus.cpython-36m-arm-linux-gnueabihf.so
```

But Python 3.6 might attempt to load this and fail.

**Comment by Gadgetoid** [comment]
What do you see if you run:

```
python --version
```

**Comment by AdrienLF** [comment]
In the end, I completely uninstall python3.6 and it solved the issue... 

I followed this guide to install python3.6 : https://gist.github.com/dschep/24aa61672a2092246eaca2824400d37f

Basically, downloading the source and building it. Since I'm quite new to linux I didn't understand it wouldn't just install it over my version of python, and caused several issues. This is why I eventually uninstalled it. 

I believe, in this case, that the issue was with my python being in /usr/local/bin/python3
